lnwire/extra_bytes.go

```diff
@@ -90,6 +90,10 @@ func (e *ExtraOpaqueData) ExtractRecords(recordProducers ...tlv.RecordProducer)
 		records = append(records, producer.Record())
 	}
 
+	// Ensure that the set of records are sorted before we attempt to
+	// decode from the stream, to ensure they're canonical.
+	tlv.SortRecords(records)
+
```

Comment:
nice! 
btw, any reason this is not done automatically within the `tlv` package before `Encode` or on `tlv.NewStream`?

Reply:
Good q...I think the intention was that the package detected an invalid ordering rather than catch it at that instance. We can def revisit that assumption tho. IIRC, we added these sorting routines once we started to display/accept the TLV fields on the RPC interface.